File: 202311/2785.go
import "slices"

func sortVowels(s string) string {
	v := map[byte]bool{
		'a': true, 'e': true, 'i': true, 'o': true, 'u': true,
		'A': true, 'E': true, 'I': true, 'O': true, 'U': true,
	}

	var vowels []byte
	n := len(s)
	for i := 0; i < n; i++ {
		if v[s[i]] {
			vowels = append(vowels, s[i])
		}
	}
	slices.Sort(vowels)

	ans := make([]byte, n)
	j := 0
	for i := 0; i < n; i++ {
		if v[s[i]] {
			ans[i] = vowels[j]
			j++
		} else {
			ans[i] = s[i]
		}
	}
	return string(ans)
}

File: 202509/2785.py
class Solution:
    def sortVowels(self, s: str) -> str:
        v = "aeiouAEIOU"
        vos = [c for c in s if c in v]
        vos.sort()
        ans = []
        i = 0
        for c in s:
            if c in v:
                ans.append(vos[i])
                i += 1
            else:
                ans.append(c)

        return "".join(ans)
